### PR TITLE
tests: testing cleanup

### LINKED ISSUE #801: integration test port numbers will overflow once reach +10, upgrade code to handle it
## Expected Behavior

-

## Current Behavior

-

## Possible Solution

-

## Steps to Reproduce

1.
2.
3.
4.

## Specification

- go version:
- gossamer version:
- gossamer commit tag:
- gossamer commit hash:
- operating system:
- additional links:

## Checklist

- [ ] I have read [CONTRIBUTING](CONTRIBUTING.md) and [CODE_OF_CONDUCT](CODE_OF_CONDUCT.md)
- [ ] I have provided as much information as possible and necessary
- [ ] I am planning to submit a pull request to fix this issue myself

### LINKED ISSUE #802: use the name of the authority when saving node rpc response to json db
This requires a map implementation (map process/pid/authority) 
related to: #801 

## Expected Behavior

-

## Current Behavior

-

## Possible Solution

-

## Steps to Reproduce

1.
2.
3.
4.

## Specification

- go version:
- gossamer version:
- gossamer commit tag:
- gossamer commit hash:
- operating system:
- additional links:

## Checklist

- [ ] I have read [CONTRIBUTING](CONTRIBUTING.md) and [CODE_OF_CONDUCT](CODE_OF_CONDUCT.md)
- [ ] I have provided as much information as possible and necessary
- [ ] I am planning to submit a pull request to fix this issue myself

### PR DESCRIPTION
**Pull request by noot**
## Changes

- move utils from tests/rpc to tests/utils
- use Node struct to keep track of process, key, port
- add to port number instead of concatenating
- move duplicate rpc test code to helper func

I'd also like to change `t.Log` in the tests to the same logger as gossamer, since `t.Log` writes all the logs after the test finishes executing, so it's harder to know what happened in what order. let me know what you guys think.

## Tests:

```
make it-rpc
make it-stress
```

## Checklist:

- [x] I have read [CONTRIBUTING](CONTRIBUTING.md) and [CODE_OF_CONDUCT](CODE_OF_CONDUCT.md)
- [x] I have provided as much information as possible and necessary
- [x] I have reviewed my own pull request before requesting a review
- [x] I have linked any relevant issues in my pull request comments
- [x] All integration tests and required coverage checks are passing

## Issues:

- closes #801 closes #802

### CODE REVIEW

File: scripts/integration-test-all.sh [hunk 1]
```diff
@@ -10,9 +10,9 @@ SLEEP_TIMEOUT=5
 TEST_QTD=3
 
 #PORT AND RPC_PORT 3 initial digits, to be concat with a suffix later when node is initialized
-PORT="700"
-RPC_PORT="854"
-IP_ADDR="0.0.0.0"
+PORT=7000
```
Comment (edwardmack) [inline scripts/integration-test-all.sh:13]: Update comment

File: tests/rpc/rpc_01-system_test.go [hunk 2]
```diff
@@ -113,41 +107,25 @@ func TestSystemRPC(t *testing.T) {
 		},
 	}
 
-	t.Log("going to start gossamer")
-
-	localPidList, err := StartNodes(t, make([]*exec.Cmd, 1))
+	t.Log("starting gossamer...")
+	nodes, err := utils.StartNodes(t, 3)
```
Comment (edwardmack) [inline tests/rpc/rpc_01-system_test.go:111]: I see you updated this to start 3 nodes here, but you've left it to start 1 node in the other tests, is there a reason for starting 3 here?

Comment (noot) [reply tests/rpc/rpc_01-system_test.go:111]: this one has an expected peers of 2, so I started 3 nodes so that the node would have 2 peers

Comment (thomasmodeneis) [reply tests/rpc/rpc_01-system_test.go:111]: This is because of the assertion for the SystemHealthResponse

**Pull request merged**